lib/lhm/chunker.rb

```diff
@@ -48,8 +47,9 @@ def bottom
       @next_to_insert
     end
 
-    def top(stride)
-      [(@next_to_insert + stride - 1), @limit].min
+    def upper_id(next_id, stride)
+      top = connection.select_value("select id from `#{ origin_name }` where id >= #{ next_id } order by id limit 1 offset #{ stride - 1}")
+      [top ? top.to_i : @limit, @limit].min
```

Comment:
This is what will happen with any real world LHM.

When you start the LHM, the table is going to have records from `X` to `Y`.
@limit is going to be `Y`.
While the LHM is running, new records are going to be inserted into the table.
Records in the range of `Y` to `Z`.

When you calculate the id boundaries for the *last* chunk that the chunker is going to copy, the query that is going to run is going to be (assuming a stride of 1000)
```
select id from ...  where id >= ... order by id limit 1 offset 999
```
That value, in the last chunk, is likely going to be higher than Y, because of the new records added while the LHM were running.
But, the records that were inserted to the table and have a primary key > `Y` had already been added to the destination via the insert trigger. We shouldn't copy them again.
Hence, if the value returned by that query (`top`) is higher than the `@limit`, then you should _only_ copy the records up to and including the `@limit`.

That's why you need the minimum of both values.